File: src/my_project/easy_problems/from101to150/relative_sort_array.py
from typing import List 
import collections

class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:


        arr = collections.Counter([x for x in arr1 if x in arr2])
        remaining = [x for x in arr1 if x not in arr2]
        # print(remaining)
        remaining.sort()
        
        # result=[]

        # print([item for item in arr2 for i in range(arr[item])] )
        
        return [item for item in arr2 for i in range(arr[item])]  + remaining


solution = Solution()

print(solution.relativeSortArray(arr1 = [2,3,1,3,2,4,6,7,9,2,19], arr2 = [2,1,4,3,9,6]))

File: src/my_project_typescript/easy_problems_typescript/from1to50/relativeSortArray.ts
function relativeSortArray(arr1: number[], arr2: number[]): number[] {

    let arr:any =  {}
    let leftArray: Array<number> = []
    let rightArray: Array<number> = []
    for (let item of arr2){
        arr[item] = 0
    }
    for (let item of arr1){
        if (arr2.includes(item) ){
            arr[item] += 1
        }
    }
    
    for (let item of arr2){
        for (let i = 0; i < arr[item]; i++){
            leftArray.push(item)
        }
    }

    for (let item of arr1){
        if (!arr2.includes(item)){
            rightArray.push(item)
        }
    }
// thiw is a new comment
let myVariable: number = 0
    rightArray = rightArray.sort((a,b) => a <=b? -1: 1)

    return leftArray.concat(rightArray)
};

console.log(relativeSortArray([2,3,1,3,2,4,6,7,9,2,19],  [2,1,4,3,9,6]))